frontend/src/App.tsx

```diff
@@ -0,0 +1,180 @@
+import React, {useState, useEffect, useMemo, FC, ComponentProps, MouseEventHandler} from "react";
+import {CustomModal as Modal} from "./components/Modal";
+import axios from "axios";
+import {TodoItem} from "./models/TodoItem";
+
+axios.defaults.xsrfCookieName = 'csrftoken'
+axios.defaults.xsrfHeaderName = 'X-CSRFToken'
+
+interface TabListProps {
+    viewCompleted: boolean;
+    displayCompleted: (status: boolean) => void;
+}
+
+const TabList: FC<TabListProps> = ({viewCompleted, displayCompleted}) => {
```

Comment:
I'll try to find the conversation for this but: as far as I remember it is a good idea to use the prop typing instead of `FC`:
```js
const TabList = (props: TabListProps) => { /**/ }
```

Reply:
Maybe @jennpeare can help us :)

Reply:
I would disagree as that leaves the return type purely inferred rather than strongly typing it. Unless we also strongly types the return type...

Reply:
Doing some research on this take, it looks like the 2 main reasons to go with the typed props approach are:
1. Children being injected as a prop attribute by FC
2. Compound Components
3. Generic Props

To point 1, the latest react releases have removed the implicit children prop.
https://github.com/DefinitelyTyped/DefinitelyTyped/commit/55dc209ceb6dbcd59c4c68cc8dfb77faadd9de12#
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/7ecb515dc7b3ea3483ed4f3de6c5784d389bed38/types/react/index.d.ts#L519-L525

As to point 2, if there is that much concern around creating the composite type, then you can drop the use of FC in that particular instance while still using it in the vast majority of cases.

I would make the same comment about point 3, components with generic props are generally rare and could skip usage of FC on an as needed basis.

Going to merge so I can keep moving forward, but will revisit this once any additional comments are made.

Reply:
The reason why we stopped using the `FC`/`FunctionComponent` is because of the children prop injection that was previously a problem. Here's an ADR about this - https://zenput.atlassian.net/wiki/spaces/DEV/pages/2343993392/ADR-0103+Use+JSX.Element+to+type+React+Function+Components

You can always add `JSX.Element` as a return type if you feel like something is not strongly typed enough, but in almost all cases if your file is `.tsx`, it will catch problematic component return types for you.

Reply:
Fair enough on the children injection, thouugh that doesn't seem to be an issue anymore.

> You can always add `JSX.Element` as a return type if you feel like something is not strongly typed enough, but in almost all cases if your file is `.tsx`, it will catch problematic component return types for you.

While this is true, leaving out the return type distances the error from the source of the issue. With a return type, the error points to the actual return line that violates the typing. Without the return type, the error is in a different component where you attempt to use the function as a component.

Reply:
Yeah the `FC` problem won't be an issue for your onboarding project, but I believe our various repos aren't on the latest version (last time I checked) where they fixed that. The ADR was a way for us to be aware of this typing quirk and help folks understand why being explicit is important. :)